cmd/cluster/context.go

```diff
@@ -393,6 +400,25 @@ func (o *contextOptions) generateContextData() (*contextData, []error) {
 		}
 	}
 
+	GetDynatraceURL := func() {
+		defer wg.Done()
+		if o.verbose {
+			fmt.Fprintln(os.Stderr, "Getting Dynatrace URL...")
+		}
+		if !strings.HasPrefix(cluster.Name(), "hs-mc-") {
+			data.DyntraceEnvURL = "cluster is Not a Management Cluster"
+			return
+		}
+
+		data.DyntraceEnvURL, err = GetDynatraceURLFromCluster(data.ClusterID)
+		if err != nil {
+			data.DyntraceEnvURL = getDynatraceURLFromRegion(cluster.Region().ID(), data.OCMEnv)
```

Comment:
Similar to other comments, I don't think we should fallback to searching by region in osdctl